File: backend/InterviewManagement.API/Models/Metadata.cs
namespace InterviewManagement.API.Models
{
    public class Metadata
    {
        public Guid Id { get; set; }

        public string Type { get; set; } = string.Empty;

        public string Value { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        // Navigation Property
        // Navigation Property
public ICollection<QuestionMetadata> QuestionMetadata { get; set; } = new List<QuestionMetadata>();
    }
}

File: frontend/interview-management-ui/src/types/metadata.ts
export interface Metadata {
  id: string;

  type: string;

  value: string;

  description: string;
}

export interface CreateMetadataRequest {
  type: string;

  value: string;

  description: string;
}

export interface UpdateMetadataRequest {
  type: string;

  value: string;

  description: string;
}